pkg/tag/generate_tag_definitions.py

```diff
@@ -1,4042 +1,66 @@
-(5000-50FF,2006)	UL	RETIRED_NumberOfSamples	1	DICOM/retired
-(5000-50FF,2008)	UL	RETIRED_SampleRate	1	DICOM/retired
-(5000-50FF,200A)	UL	RETIRED_TotalTime	1	DICOM/retired
-(5000-50FF,200C)	ox	RETIRED_AudioSampleData	1	DICOM/retired
-(5000-50FF,200E)	LT	RETIRED_AudioComments	1	DICOM/retired
-(5000-50FF,2500)	LO	RETIRED_CurveLabel	1	DICOM/retired
-(5000-50FF,2600)	SQ	RETIRED_CurveReferencedOverlaySequence	1	DICOM/retired
-(5000-50FF,2610)	US	RETIRED_CurveReferencedOverlayGroup	1	DICOM/retired
-(5000-50FF,3000)	ox	RETIRED_CurveData	1	DICOM/retired
-(6000-60FF,0012)	US	RETIRED_OverlayPlanes	1	DICOM/retired
-(6000-60FF,0052)	US	RETIRED_OverlayPlaneOrigin	1	DICOM/retired
-(6000-60FF,0060)	CS	RETIRED_OverlayCompressionCode	1	DICOM/retired
-(6000-60FF,0061)	SH	RETIRED_OverlayCompressionOriginator	1	DICOM/retired
-(6000-60FF,0062)	SH	RETIRED_OverlayCompressionLabel	1	DICOM/retired
-(6000-60FF,0063)	CS	RETIRED_OverlayCompressionDescription	1	DICOM/retired
-(6000-60FF,0066)	AT	RETIRED_OverlayCompressionStepPointers	1-n	DICOM/retired
-(6000-60FF,0068)	US	RETIRED_OverlayRepeatInterval	1	DICOM/retired
-(6000-60FF,0069)	US	RETIRED_OverlayBitsGrouped	1	DICOM/retired
-(6000-60FF,0110)	CS	RETIRED_OverlayFormat	1	DICOM/retired
-(6000-60FF,0200)	US	RETIRED_OverlayLocation	1	DICOM/retired
-(6000-60FF,0800)	CS	RETIRED_OverlayCodeLabel	1-n	DICOM/retired
-(6000-60FF,0802)	US	RETIRED_OverlayNumberOfTables	1	DICOM/retired
-(6000-60FF,0803)	AT	RETIRED_OverlayCodeTableLocation	1-n	DICOM/retired
-(6000-60FF,0804)	US	RETIRED_OverlayBitsForCodeWord	1	DICOM/retired
-(6000-60FF,1100)	US	RETIRED_OverlayDescriptorGray	1	DICOM/retired
-(6000-60FF,1101)	US	RETIRED_OverlayDescriptorRed	1	DICOM/retired
-(6000-60FF,1102)	US	RETIRED_OverlayDescriptorGreen	1	DICOM/retired
-(6000-60FF,1103)	US	RETIRED_OverlayDescriptorBlue	1	DICOM/retired
-(6000-60FF,1200)	US	RETIRED_OverlaysGray	1-n	DICOM/retired
-(6000-60FF,1201)	US	RETIRED_OverlaysRed	1-n	DICOM/retired
-(6000-60FF,1202)	US	RETIRED_OverlaysGreen	1-n	DICOM/retired
-(6000-60FF,1203)	US	RETIRED_OverlaysBlue	1-n	DICOM/retired
-(6000-60FF,4000)	LT	RETIRED_OverlayComments	1	DICOM/retired
-(7FE0,0020)	OW	RETIRED_CoefficientsSDVN	1	DICOM/retired
-(7FE0,0030)	OW	RETIRED_CoefficientsSDHN	1	DICOM/retired
-(7FE0,0040)	OW	RETIRED_CoefficientsSDDN	1	DICOM/retired
-(7F00-7FFF,0010)	ox	RETIRED_VariablePixelData	1	DICOM/retired
-(7F00-7FFF,0011)	US	RETIRED_VariableNextDataGroup	1	DICOM/retired
-(7F00-7FFF,0020)	OW	RETIRED_VariableCoefficientsSDVN	1	DICOM/retired
-(7F00-7FFF,0030)	OW	RETIRED_VariableCoefficientsSDHN	1	DICOM/retired
-(7F00-7FFF,0040)	OW	RETIRED_VariableCoefficientsSDDN	1	DICOM/retired
-#
-# EOF
-#
-"""
+func maybeInitTagDict() {{
+	if len(tagDict) > 0 {{
+		return
+	}}
+	tagDict = make(map[Tag]Info)
+{new_line.join(f'	tagDict[{t.keyword}] = Info{{{t.keyword}, []string{{{", ".join(map(wrap_in_quotes, t.vr))}}}, "{t.name}", "{t.keyword}", "{t.vm}", {str(t.retired).lower()}}}' for t in tags)}
```

Comment:
nit: IMO this isn't the most readable. Would it help to factor out this population of tagDict perhaps using a string builder (StringIO) and a for loop through the tags? We don't have to block on it though. 

Also, I wonder if instead of adding the entries to the map if this would be better as constructing a giant map literal in Go (e.g. https://go.dev/tour/moretypes/20) (this is another legacy thing). We can explore that in a follow up PR.

Reply:
Changed the use the giant map literal and I think it looks good now.